packages/dds/tree/src/test/shared-tree/testTree.ts

```diff
@@ -139,34 +126,36 @@ export class TestTree {
 	}
 
 	/**
-	 * Runs the given `command`, applying the resulting edit to the document.
+	 * Runs the given `transaction`, applying the resulting edit to the document.
+	 * @param transaction - the transaction to run. It may not be aborted.
 	 * @returns A edit that can be sequenced by the `Sequencer`.
 	 */
-	public runTransaction(command: SucceedingCommand): TestTreeEdit {
-		const trueCommand = commandWithResult(command);
-		let changeset: DefaultChangeset | undefined;
-		const checkout = {
-			forest: this.forest,
-			changeFamily: defaultChangeFamily,
-			submitEdit: (change: DefaultChangeset): void => {
-				changeset = change;
-			},
-		};
-		const result = runSynchronousTransaction(checkout, trueCommand);
-		assert(
-			result === TransactionResult.Apply && changeset !== undefined,
-			"The transaction should result in an edit being submitted",
-		);
+	public runTransaction(
+		transaction: (forest: IForestSubscription, editor: DefaultEditBuilder) => void,
+	): TestTreeEdit {
+		const editor = defaultChangeFamily.buildEditor((edit) => {
+			const delta = defaultChangeFamily.intoDelta(edit);
+			this.forest.applyDelta(delta);
+		}, this.forest.anchors);
+		transaction(this.forest, editor);
+		const changes = editor.getChanges();
+
+		// Using anonymous changes makes it impossible to chronologically order them during composition.
+		// Such an ordering is needed when composing/squashing inverse changes with other changes, which is currently
+		// not expected to happen in transactions but that could change in the future.
+		const anonChanges = changes.map((c) => makeAnonChange(c));
+
+		const changeset = defaultChangeFamily.rebaser.compose(anonChanges);
 		const revision = mintRevisionTag();
-		const delta = this.editManager.addLocalChange(revision, changeset);
+		// Forest and anchors already reflect the results of this change since they were updated during transaction.
```

Comment:
These changes are problematic in that they introduce a divergence between how SharedTree transactions work and how TestTree transactions work.

We need to have a conversation about how to avoid having this code in the first place, meaning that the tests should leverage the production code, but I' rather not make things worse in the meantime.

It should be possible to update this code so that it still mimics what happens in the production code.

Reply:
Before this PR this test this code would apply the changes, roll them back, then submit+reapply them using a bunch of test only code. My understanding is SharedTree doesn't do this for its editing APIs anymore? This PR makes this test no longer take that approach.

Reply:
You're right. I had no idea!